### PR TITLE
Refactor PATH_BIT_DRAW_TYPE_LIGHTS to use strong enum

### LINKED ISSUE #14817: Refactor PATH_BIT_DRAW_TYPE to use strong enum
```
enum
{
    PATH_BIT_DRAW_TYPE_LIGHTS,
    PATH_BIT_DRAW_TYPE_BINS,
    PATH_BIT_DRAW_TYPE_BENCHES,
    PATH_BIT_DRAW_TYPE_JUMPING_FOUNTAINS
};
```
Should be refactored to an `enum class PathBitDrawType : uint8_t`.

### PR DESCRIPTION
**Pull request by Sciencentistguy**
Closes #14817

### CODE REVIEW

File: compile_commands.json [hunk 1]
```diff
@@ -0,0 +1 @@
+build/compile_commands.json
```
Comment (duncanspumpkin) [inline compile_commands.json:1]: spurious new file? Delete

File: src/openrct2/world/Scenery.h [hunk 2]
```diff
@@ -188,12 +189,12 @@ enum
...
+enum class PathBitDrawType : uint8_t
 {
-    PATH_BIT_DRAW_TYPE_LIGHTS,
-    PATH_BIT_DRAW_TYPE_BINS,
-    PATH_BIT_DRAW_TYPE_BENCHES,
-    PATH_BIT_DRAW_TYPE_JUMPING_FOUNTAINS
+    Lights,
+    Bins,
+    Benches,
+    Jumping_Fountains
```
Comment (duncanspumpkin) [inline src/openrct2/world/Scenery.h:197]: Can you remove the underscore and all of these should really be singular as they represent a single object.

Comment (Sciencentistguy) [reply src/openrct2/world/Scenery.h]: Done

File: compile_commands.json (hunk 1 above)
Comment (Sciencentistguy) [reply compile_commands.json]: Whoops. Force pushed that out.

File: src/openrct2/world/Scenery.h [hunk 3]
```diff
@@ -181,12 +182,13 @@ enum
...
 {
-    PATH_BIT_DRAW_TYPE_LIGHTS,
-    PATH_BIT_DRAW_TYPE_BINS,
-    PATH_BIT_DRAW_TYPE_BENCHES,
-    PATH_BIT_DRAW_TYPE_JUMPING_FOUNTAINS
+    Light,
+    Bin,
+    Bench,
+    JumpingFountain
+
```
Comment (duncanspumpkin) [inline src/openrct2/world/Scenery.h:191]: Could you delete this white space and end the fountain entry with a comma for the unlikely event that we are adding more to the enum

**Review by duncanspumpkin — APPROVED** [review]

File: src/openrct2/world/Scenery.h (hunk 3 above)
Comment (Sciencentistguy) [reply src/openrct2/world/Scenery.h]: Done

**Comment by duncanspumpkin** [comment]
Good to merge when ci's turn green

**Pull request merged**